lib/modules.nix

```diff
@@ -153,8 +154,74 @@ rec {
             # a `_module.args.pkgs = import (fetchTarball { ... }) {}` won't
             # start a download when `pkgs` wasn't evaluated.
             type = types.lazyAttrsOf types.raw;
-            internal = true;
-            description = "Arguments passed to each module.";
+            # Only render documentation once at the root of the option tree,
+            # not for all individual submodules.
+            internal = prefix != [];
+            example = literalExpression ''
+              {
+                unstable = import nixpkgs {};
+                myLib = { addOne = x: x + 1; };
+              }
+            '';
+            # TODO: Change the type of this option to a submodule with a
+            # freeformType, so that individual arguments can be documented
+            # separately
+            description = ''
+              Additional arguments passed to each module in addition to ones
+              like <literal>lib</literal>, <literal>config</literal>,
+              <literal>pkgs</literal>, <literal>modulesPath</literal> for
```

Comment:
> These are NixOS-specific and don't belong in the generated documentation of other configuration systems. They are better documented as freeform options.

I should reword this sentence (there's a `for nixpkgs` afterwards, but it's very ambiguous). I think it's valid to mention `pkgs` and `modulesPath` in this documentation because there's no other place where these are documented currently, and NixOS modules are by far the most common use case. With a freeform type we should be able to make this documentation NixOS-specific in the future though.

Reply:
I just dropped the `for nixpkgs`. This sentence specifically says "ones like ...", which indicates that it's just an example of some arguments the reader might be familiar with. I think this should be fine